connect_deprish_to_ferc1.py

```diff
@@ -503,12 +532,177 @@ def connect(self):
+
+        Args:
+            same_smol (pandas.DataFrame): table with matched records from ferc1
+                and depreciation records.
+            data_col (string): name of the ferc1 data column.
+            idx_cols_ferc1 (list): columns to group by.
+            split_cols (list): ordered list of columns to split porportionally
+                based on. Ordered based on priority: if non-null result from
+                frist column, result will include first column result, then
+                second and so on.
+        Returns:
+            pandas.DataFrame: a modified version of `same_smol` with a new
+                assigned data_col
+
+        """
+        # we want to know the sum of the potential split_cols for each ferc1
+        # option
+        df_fgb = (
+            same_smol[idx_cols_ferc1 + split_cols]
+            .fillna(0)  # remove w/ pandas 1.1
+            .groupby(by=idx_cols_ferc1)  # add dropna=False w/ pandas 1.1
+            .sum()
+            .reset_index()
+        )
+        df_w_tots = (
+            pd.merge(same_smol, df_fgb,
+                     on=idx_cols_ferc1,
+                     suffixes=("", "_fgb"))
+        )
+        # for each of the columns we want to split the frc data by
+        # generate the % of the total group, so we can split the data_col
+        # we want an iterater here w/ n at the end of the column names
+        # so we can later sort in order of priority
+        n = 1
+        for split_col in split_cols:
+            df_w_tots[f"{split_col}_pct"] = (
+                df_w_tots[split_col] / df_w_tots[f"{split_col}_fgb"])
+            df_w_tots[f"{data_col}_ferc1_deprish_split_{n}"] = (
+                df_w_tots[data_col] * df_w_tots[f"{split_col}_pct"])
+            n = n + 1
+
+        new_data_col = self._get_clean_new_data_col(
+            f"{data_col}_ferc1_deprish")
+        # now we have several ways to split the ferc data. we want to choose
+        # the first non-null option.
+        df_w_tots[new_data_col] = pd.NA
+        fill_col_options = set(
+            df_w_tots.filter(like=f"{data_col}_ferc1_deprish_split_").columns)
+        for fill_col in fill_col_options:
+            df_w_tots[new_data_col] = (
+                df_w_tots[new_data_col].fillna(df_w_tots[fill_col]))
+
+        # merge in the newly generated split/assigned data column
+        df_final = pd.merge(
+            same_smol, df_w_tots[
+                idx_cols_ferc1 + ['record_id_eia_deprish', new_data_col]])
+        return df_final
+
+    def _get_clean_new_data_col(self, data_col):
+        # some of the data columns already have a ferc1 suffix because the same
+        # column name also shows up in the EIA data... so we want to remove the
+        # double ferc1 if it shows up
+        new_data_col = f"{data_col}_ferc1_deprish"
+        if "ferc1_ferc1" in new_data_col:
+            new_data_col = new_data_col.replace("ferc1_ferc1", "ferc1")
+        return new_data_col
```

Comment:
Do you mean always run `new_data_col = new_data_col.replace("ferc1_ferc1", "ferc1")`? I guess I could do that. The replace with `""` wouldn't be ideal because I want the `_ferc1_deprish` as the suffix.

To the later point, yes and that was an error. Already fixed it.

Reply:
Yeah I meant replace double ferc1 with single ferc1, not an empty string. whoops.